firmware/controllers/idle_thread.cpp

```diff
@@ -170,6 +172,13 @@ percent_t getIdlePosition(void) {
 static float autoIdle() {
 	percent_t tpsPos = getTPS(PASS_ENGINE_PARAMETER_SIGNATURE);
 	if (tpsPos > boardConfiguration->idlePidDeactivationTpsThreshold) {
+		// Don't store old I and D terms if PID doesn't work anymore.
+		// Otherwise they will affect the idle position much later, when the throttle is closed.
+		if (mightResetPid) {
+			mightResetPid = false;
+			shouldResetPid = true;
```

Comment:
This is a main idea of 'mightResetPid': We need to reset PID when the throttle is opened and PID deactivated.
Otherwise they will affect the idle position much later, when the throttle is closed.
BUT. We can't just set shouldResetPid immediately - or it will be reseted every time, every cycle.
Instead we wait until at least 1 PID iteration works - and set mightResetPid, a "permission" to do a reset.